Lesson18/deniz/script.js

```diff
@@ -0,0 +1,192 @@
+      newArray.push(array[i]); 
+      // 'a' !== 'b' => ['a']
+      // 'b' === 'b' => ['a']
+      // 'c' !== 'b' => ['a', 'c']
+    }
+  }
+
+  return newArray;
+}
+
+console.log("exercise 6:",removeValueFromArray(testArrayEx6, 3), removeValueFromArray(['a', 'b', 'c'], 'b'));
+console.log('----------------------------');
+// 7. Write a function that takes an object and returns the keys as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["name", "age", "city"]
+const testObjEx7 = {name: "John", age: 30, city: "New York", isStudent: false};
+console.log(Object.keys(testObjEx7));
+
+console.log("exercise 7:");
+console.log('----------------------------');
+// 8. Write a function that takes an object and returns the values as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["John", 30, "New York"]
+const testObjEx8 = {name: "John", age: 30, city: "New York"};
+
+
+console.log("exercise 8:", Object.values(testObjEx8));
+console.log('----------------------------');
+// 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
+// Input: ({name: "John", age: 30, city: "New York"}, "age")
+// Output: true
+// Input: ({name: "John", age: 30, city: "New York"}, "surname")
+// Output: false
+const testObjEx9 = {name: "John", age: 30, city: "New York", number: undefined};
+
+function ex9 (object, key) {
+return object.hasOwnProperty(key);
+}
+
+
+
+console.log("Ex 9:", ex9(testObjEx9, 'age'),
+ex9(testObjEx9, "surname"));
+
+console.log('----------------------------');
+// 10. Write a function that takes an object and a key-value pair, and adds the key-value pair to the object.
+// Input: ({name: "John", age: 30}, "city", "New York")
+// Output: {name: "John", age: 30, city: "New York"}
+const testObjEx10 = {name: "John", age: 30};
+
+function ex10 (object, key, value) {
+  object[key] = value;
+  return object;
+}
+console.log("exemple 10:", ex10(testObjEx10, "city", "New York"));
+console.log('----------------------------');
+// 11. Write a function that takes an object and a key, and removes the key-value pair from the object.
+// Input: ({name: "John", age: 30, city: "New York"}, "city")
+// Output: {name: "John", age: 30}
+const testObjEx11 = {name: "John", age: 30, city: "New York"};
+
+function removeKeyValue (object, key) {
+  delete object[key];
+  return object;
+}
+
+
+console.log("exemple 11:", removeKeyValue(testObjEx11, "city"));
+console.log('----------------------------');
+// 12. Write a function that takes two arrays of equal length and returns an object with the first array as keys and the second array as values.
+// Input: (["name", "age", "city"], ["John", 30, "New York"])
+// Output: {name: "John", age: 30, city: "New York"}
+const array1Ex12 = ["name", "age", "city"];
+const array2Ex12 = ["John", 30, "New York"];
+
+function arraysToObject (keys, values) {
+let obj = [];
```

Comment:
```suggestion
let obj = {};
```

Reply:
Also, it would be good to check that the length of 'keys' and 'values' array is equal as the first step in this function.